src/hb-array.hh

```diff
@@ -196,6 +196,15 @@ struct hb_array_t : hb_iter_with_fallback_t<hb_array_t<Type>, Type&>
   const T *as () const
   { return length < hb_null_size (T) ? &Null (T) : reinterpret_cast<const T *> (arrayZ); }
 
+  template <typename T,
+	    unsigned P = sizeof (Type),
+	    hb_enable_if (P == 1)>
+  bool in_range (const T *p, unsigned int size = T::static_size) const
+  {
+    return ((const char *) p) >= arrayZ
+	&& ((const char *) p + size) <= arrayZ + length;
+  }
+
```

Comment:
The addition to size can overflow.  I know not big deal but good to check.  Check how we do in sanitizer.

Reply:
Can also maybe enable if byte array or if T is_same as array member type...

Reply:
> Check how we do in sanitizer.

Oh, worth to have, there is an assert and a sign check there, can you please indicate which line you are referring. Thanks

Reply:
Maybe also rename to `check_range`.
```c++
  bool check_range (const void *base,
                    unsigned int len) const
  {
    const char *p = (const char *) base;
    bool ok = !len ||
              (this->start <= p &&
               p <= this->end &&
               (unsigned int) (this->end - p) >= len &&
               this->max_ops-- > 0);

    DEBUG_MSG_LEVEL (SANITIZE, p, this->debug_depth+1, 0,
                     "check_range [%p..%p]"
                     " (%d bytes) in [%p..%p] -> %s",
                     p, p + len, len,
                     this->start, this->end,
                     ok ? "OK" : "OUT-OF-RANGE");

    return likely (ok);
  }
```